src/components/ui/form/switch/Switch.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { ChangeEvent, useCallback } from 'react';
+import styled, { css } from 'react-emotion';
+import { compose, setDisplayName } from 'recompose';
+import { RequiredInputComponentProps } from '../../../RequiredComponentProps';
+import {
+  withComponentTheme,
+  WithInnerComponentThemeProps,
+} from '../../../util/enhancers';
+import { Icon } from '../../icon';
+import { SwitchTheme } from './SwitchTheme';
+
+export interface SwitchProps
```

Comment:
You can extend ValueOnChangeProps<boolean>.
Not sure if this is used for all form components, but it should be, to enforce consistent API.